components/ai_chat/core/browser/engine/engine_consumer_oai.cc

```diff
@@ -113,8 +113,8 @@ base::Value::List BuildMessages(
 EngineConsumerOAIRemote::EngineConsumerOAIRemote(
     const mojom::Model& model,
     scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
-    ModelService* model_service)
-    : model_options_(*model.options->get_custom_model_options()) {
+    std::shared_ptr<ModelService> model_service)
```

Comment:
I think shared_ptr is banned - could we use scoped_refptr instead?
https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#std_shared_ptr-banned

Reply:
I reverted this logic actually; wasn't a fan of modifying the constructor.